test/src/sdk-harness/test_projects/token_ops/src/main.sw

```diff
@@ -53,6 +53,13 @@ impl TestFuelCoin for Contract {
     }
 
     fn send_message(recipient: b256, msg_data: Vec<u64>, coins: u64) {
-        send_message(recipient, msg_data, coins);
+        let mut data = Bytes::new();
+        if msg_data.len() > 0 {
+            data.push(msg_data.get(0).unwrap());
+            data.push(msg_data.get(1).unwrap());
+            data.push(msg_data.get(2).unwrap());
```

Comment:
Why do we hard code an assumption of 3 data points instead of looping over any amount of data in the vec?

Reply:
Unrolled loop for extreme test efficiency :smile:

Reply:
No good reason. The assumption of 3 data points in a contract which only exists for testing was good enough to get the job done but could be made more robust with a while loop.